### PR TITLE
[45] [Backend] As a user, I can logout

### PR DESCRIPTION
**Pull request by manh-t**
https://github.com/manh-t/flutter-survey/issues/45

## What happened 👀

Call POST `/api/v1/oauth/revoke`

## Proof Of Work 📹

```
/flutter (11816): *** Request ***
I/flutter (11816): uri: https://nimble-survey-web-staging.herokuapp.com/api/v1/oauth/revoke
I/flutter (11816): method: POST
I/flutter (11816): responseType: ResponseType.json
I/flutter (11816): followRedirects: true
I/flutter (11816): connectTimeout: 30000
I/flutter (11816): sendTimeout: 0
I/flutter (11816): receiveTimeout: 30000
I/flutter (11816): receiveDataWhenStatusError: true
I/flutter (11816): extra: {}
I/flutter (11816): headers:
I/flutter (11816):  content-type: application/json; charset=utf-8
I/flutter (11816): data:
I/flutter (11816): {token: ul_JwBWFf9e5n3a47auLRRx3RuthEjEgnwbOBg9vxAc, client_id: z9iUamZLvRgtVVtRJ8UqItg2vmncGyEi30p1eWEddnA, client_secret: 1vqRNMxq-Yx83A61GNjLb17qxCGKxHDb8EmB3MKdxqA}
I/flutter (11816): *** Response ***
I/flutter (11816): uri: https://nimble-survey-web-staging.herokuapp.com/api/v1/oauth/revoke
I/flutter (11816): statusCode: 200
I/flutter (11816): headers:
I/flutter (11816):  connection: keep-alive
I/flutter (11816):  cache-control: max-age=0, private, must-revalidate
I/flutter (11816):  transfer-encoding: chunked
I/flutter (11816):  date: Fri, 04 Jun 2021 09:39:59 GMT
I/flutter (11816):  vary: Accept-Encoding, Origin
I/flutter (11816):  content-encoding: gzip
I/flutter (11816):  strict-transport-security: max-age=31536000; includeSubDomains
I/flutter (11816):  referrer-policy: strict-origin-when-cross-origin
I/flutter (11816):  x-permitted-cross-domain-policies: none
I/flutter (11816):  content-type: application/json; charset=utf-8
I/flutter (11816):  x-xss-protection: 1; mode=block
I/flutter (11816):  server: Cowboy
I/flutter (11816):  x-request-id: d33edb1b-43d0-4e58-80e6-9b20303cb4e7
I/flutter (11816):  x-download-options: noopen
I/flutter (11816):  x-runtime: 0.016876
I/flutter (11816):  etag: W/"5c7e82ca4fb17519e2a01ee8934097e1"
I/flutter (11816):  via: 1.1 vegur
I/flutter (11816):  x-frame-options: SAMEORIGIN
I/flutter (11816):  x-content-type-options: nosniff
I/flutter (11816): Response Text:
I/flutter (11816): {}

```

### CODE REVIEW

File: lib/api/local/shared_preference_helper.dart [hunk 1]
```diff
@@ -32,6 +34,9 @@ class SharedPreferenceHelperImpl extends SharedPreferenceHelper {
     return (tokenType + " " + authToken).trim();
   }
 
+  @override
+  String get rawAuthToken => _sharedPreference.getString(AUTH_TOKEN) ?? "";
```
Comment (markgravity) [inline lib/api/local/shared_preference_helper.dart:38]: I think we should update `AUTH_TOKEN` also to match with the new name `rawAuthToken`.
It makes less confuse 🤔

File: lib/api/local/shared_preference_helper.dart [hunk 2]
```diff
@@ -28,23 +30,26 @@ class SharedPreferenceHelperImpl extends SharedPreferenceHelper {
   @override
   String get authToken {
     final tokenType = _sharedPreference.getString(TOKEN_TYPE) ?? "";
-    final authToken = _sharedPreference.getString(AUTH_TOKEN) ?? "";
+    final authToken = _sharedPreference.getString(RAW_AUTH_TOKEN) ?? "";
```
Comment (markgravity) [inline lib/api/local/shared_preference_helper.dart:33]: I think you can re-use your defined property `rawAuthToken`. What do you think? 🤔

**Review by markgravity — APPROVED** [review]
LGTM with a suggestion 👍

File: lib/api/local/shared_preference_helper.dart [hunk 3]
```diff
@@ -28,23 +30,26 @@ class SharedPreferenceHelperImpl extends SharedPreferenceHelper {
   @override
   String get authToken {
     final tokenType = _sharedPreference.getString(TOKEN_TYPE) ?? "";
-    final authToken = _sharedPreference.getString(AUTH_TOKEN) ?? "";
-    return (tokenType + " " + authToken).trim();
+    final rawAuthToken = _sharedPreference.getString(RAW_AUTH_TOKEN) ?? "";
+    return (tokenType + " " + rawAuthToken).trim();
```
Comment (sleepylee) [inline lib/api/local/shared_preference_helper.dart:34]: shouldn't we do string format as: `"$tokenType $rawAuthToken`? 🤔  I think `trim()` is also unnecessary.

**Comment by manh-t** [comment]
@sleepylee Rebased and updated 🚀

**Review by sleepylee — APPROVED** [review]

**Pull request merged**